ArduPlane/servos.cpp

```diff
@@ -342,6 +342,33 @@ void Plane::set_servos_manual_passthrough(void)
     }
 }
 
+/*
+  Scale the throttle to conpensate for battery voltage drop
+ */
+void Plane::throttle_voltage_comp()
+{
+    // return if not enabled, or setup incorrectly
+    if (g2.fwd_thr_batt_voltage_min >= g2.fwd_thr_batt_voltage_max || !is_positive(g2.fwd_thr_batt_voltage_max)) {
+        return;
+    }
+
+    float batt_voltage_resting_estimate = AP::battery().voltage_resting_estimate(g2.fwd_thr_batt_idx);
+    // Return for a very low battery
+    if (batt_voltage_resting_estimate < 0.25f * g2.fwd_thr_batt_voltage_min) {
+        return;
+    }
+
+    // constrain read voltage to min and max params
+    batt_voltage_resting_estimate = constrain_float(batt_voltage_resting_estimate,g2.fwd_thr_batt_voltage_min,g2.fwd_thr_batt_voltage_max);
+
+    // Scale the throttle up to compensate for voltage drop
+    // Ratio = 1 when voltage = votable max, ratio increases as voltage drops
+    const float ratio = g2.fwd_thr_batt_voltage_max / batt_voltage_resting_estimate;
+
+    SRV_Channels::set_output_scaled(SRV_Channel::k_throttle,
```

Comment:
I think this will break the throttle slew rate code

Reply:
I have double checked, slew rate is applied after this so it is still working as expected. This done as part of [set_servos_controlled](https://github.com/ArduPilot/ardupilot/blob/ce68e0fbb5c8cc5fe9cf894c3e477c824e1f3995/ArduPlane/servos.cpp#L699) and the slew rate is applied [few lines after](https://github.com/ArduPilot/ardupilot/blob/ce68e0fbb5c8cc5fe9cf894c3e477c824e1f3995/ArduPlane/servos.cpp#L715)

Reply:
I have looked for a parameter in the latest FW that will set the throttle scaling factor. I can't find anything in the 4.06 FW that seems to include the voltage compensation function. Am I missing the point?

The #10622 seems to be pointing here...

thx, Mike

Reply:
@mikeo0 Its not in the stable release yet, if you want to try it its in the dev branch (4.1) https://firmware.ardupilot.org/Plane/latest/

It is the dev branch, not stable, there are other changes also, take care.